src/frontend/src/lib/components/tokens/TokensDisplayHandler.svelte

```diff
@@ -19,22 +20,23 @@
 
 	const parseTokenKey = (token: Token) => `${token.id.description}-${token.network.id.description}`;
 
-	let tokenKeys: string;
-	$: tokenKeys = tokens?.map(parseTokenKey).join(',') ?? '';
-
-	let sortedTokensKeys: string;
-	$: sortedTokensKeys = sortedTokens?.map(parseTokenKey).join(',') ?? '';
+	let sortedTokensKeys: TokenIndexKey[];
+	$: sortedTokensKeys = sortedTokens.map(parseTokenKey);
 
 	const updateTokensToDisplay = () => {
 		if (!$pointerEventStore) {
 			// No pointer events, so we are not worried about possible glitches on user's interaction.
+			tokensUiStore.set({ tokensKeys: sortedTokensKeys });
```

Comment:
Is there a particular reason why you implemented the assignement in an imperative way instead of keeping the original reactive assignement?

i.e. why not something like?

```
$: tokensUiStore.set({ tokensKeys: tokens?.map(parseTokenKey)
```

Reply:
no particular reason: i am reading it imperatively, since i wanted to avoid recursive loops; so, I wanted to set it imperatively too, to have all concentrated in this function.

But no strong opinion. Maybe like this, there will be slightly less assignment iterations (it is not assigned for every change of tokens, but only on re-sorting)?

If you prefer I put it reactively

Reply:
Thing is we are doing two things in this PR: introducing a store plus modifying the behavior. Given that we do not have test around this, I would stick to existing logic.

In addition, we are setting back to back two different states in the function - `tokensUiStore.set` and `tokens = sortedTokens;` - which make me ask myself if it's good or not.

Long story short, this might good be an improvement but I would split the feature in two and when implementing this change, do some manual testing and reporting to check if it behaves the same, better or worse.

Makes sense?

Reply:
Let's do like this:

Since this was already a split of PR #2152, I will first do a PR for the other part that involves adding tests to the function that updates the tokens to display: PR  https://github.com/dfinity/oisy-wallet/pull/2171

Maybe that would be enough, so that this one is not needed anymore